doc/example/gradient_check.ipynb

```diff
@@ -0,0 +1,894 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Gradient checks\n",
+    "\n",
+    "It is best practice to do gradient checks before and after gradient-based optimization.\n",
```

Comment:
I think it would be good to include some rationale for why to check it afterwards and what to look for. I.e. except for parameters with active bounds, the values should be close to 0. At the same time, this might make it difficult to get good FD approximations.